vcd/resource_vcd_vapp_vm_test.go

```diff
@@ -466,6 +468,133 @@ resource "vcd_vm" "boot-image-vm" {
 }
 `
 
+// TestAccVcdVm_WithoutIopsRights tests VM creation with a user that does not have any View/edit IOPS rights.
+// This test will panic without fix made in https://github.com/vmware/terraform-provider-vcd/pull/1312
+func TestAccVcdVm_WithoutIopsRights(t *testing.T) {
+	preTestChecks(t)
+	skipIfNotSysAdmin(t)
+
+	var params = StringMap{
+		"ProviderVcdSystem": providerVcdSystem,
+		"ProviderVcdOrg":    providerVcdOrg1,
+		"Org":               testConfig.VCD.Org,
+		"Name":              t.Name(),
+		"Vdc":               testConfig.Nsxt.Vdc,
+		"Catalog":           testConfig.VCD.Catalog.NsxtBackedCatalogName,
+		"CatalogItem":       testConfig.VCD.Catalog.NsxtCatalogItem,
+		"Tags":              "vapp vm",
+	}
+	testParamsNotEmpty(t, params)
+
+	configText := templateFill(testAccCheckVcdVmWithoutIopsRights, params)
+	if vcdShortTest {
+		t.Skip(acceptanceTestsSkipped)
+		return
+	}
+
+	vcdClient := createTemporaryVCDConnection(false)
+	// Save the rights that are manipulated to restore them when the test is finished
+	var modifiedRights []types.OpenApiReference
+
+	// Restore the role with the rights that were removed, if needed, at the end of the test
+	defer func() {
+		if len(modifiedRights) > 0 {
+			role, err := vcdClient.Client.GetGlobalRoleByName("Organization Administrator")
+			if err != nil {
+				t.Fatalf("could not restore rights after %s test finished, %s", t.Name(), err)
+			}
+			err = role.AddRights(modifiedRights)
+			if err != nil {
+				t.Fatalf("could not restore rights after %s test finished, %s", t.Name(), err)
+			}
+		}
+	}()
+
+	debugPrintf("#[DEBUG] CONFIGURATION: %s\n", configText)
+	resource.Test(t, resource.TestCase{
+		ProviderFactories: buildMultipleProviders(),
+		CheckDestroy:      testAccCheckVcdStandaloneVmDestroy(t.Name(), params["Org"].(string), params["Vdc"].(string)),
+		Steps: []resource.TestStep{
+			{
+				Config: configText,
+				PreConfig: func() {
+					role, err := vcdClient.Client.GetGlobalRoleByName("Organization Administrator")
+					if err != nil {
+						if govcd.ContainsNotFound(err) {
+							t.Skipf("could not find required global role Organization Administrator")
+						}
+						t.Fatal(err)
+					}
+					filter := make(url.Values)
+					filter.Set("filter", "name==Organization vDC Disk*")
```

Comment:
Nit/request to clarify. The right here is called "Organization vDC Disk", yet the function comment doesn't mention that - only "IOPS rights". Hence, it's not very easy to follow the function through when reading without context.

Maybe you could extend/align the right name in function description more?

Reply:
Done, changed the test name with full right name